tests/framework/resource/test_resource_group.py

```diff
@@ -12,29 +12,41 @@
 from vivarium.framework.values import AttributePipeline, Pipeline, ValueModifier, ValueSource
 
 
-def test_resource_group(mocker: MockerFixture) -> None:
+class TestResourceGroup:
```

Comment:
The diff is hard to read here. The mains are 
(1) Instead of creating a list[Resource] for the resources, I parameterize and create either a list[Column] or a single Resource
(2) moved the test into a TestResourceGroup for the sole purpose of leveraging the class `component` attribute (in order to ensure the same component is used throughout)